libs/preprocessor/CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ set(headers
   ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/util/detail/pp/strip_parens.hpp
 )
 
-add_library(hpx_preprocessor STATIC ${headers})
+add_library(hpx_preprocessor ${hpx_libs_link_mode} ${headers})
```

Comment:
Shouldn't modules always be linked statically?

Reply:
Need not after this change... ;) We could keep it as it was, but not sure if there's any reason to restrict it to always be linked statically.

I'll revert this for now so we don't end up with any funny surprises from this for the release. Statically linking is pretty safe...

Reply:
@msimberg My only concern would be that we end up with dozens of very small shared libraries that will (possibly) significantly slow down startup on large distributed runs.

Reply:
Ok, that could actually end up being a problem. So definitely staying with static for now. Thanks for noticing!